Add support autofix with `checkAgainstRule` (#6466)
stylelint/stylelint: docs/developer-guide/plugins.md

- `ruleSettings`: settings for the rule you are invoking
- `root`: the root node to run this rule against
- `result?`: the PostCSS result for resolving and invoking custom rules

> - `context?`: the PluginContext for the rule you are invoking

Reviewer: [suggest] Since `PluginContext` is an internal name, could you change it like this?

Also, what about adding a link to the document that explains the `context` object?

```suggestion
- `context?`: the [context](rules.md#add-autofix) for the rule you are invoking
```

Author: Great suggestion! Definitely an improvement to the information architecture and developer experience.

  /* .. */
];

> function myPluginRule(primaryOption, secondaryOptionObject, pluginContext) {

Reviewer: [suggest] I suggest also changing here for consistency:

```suggestion
function myPluginRule(primaryOption, secondaryOptionObject, ruleContext) {
```

        ruleName: "at-rule-no-unknown",
        ruleSettings: [primaryOption, defaultedOptions],
        root: postcssRoot,
        result: postcssResult,

>         context: pluginContext

Reviewer: ```suggestion
        context: ruleContext
```
